### PR TITLE
feat(legi-shared): adding option to show select dropdown icon for aut…

### PR DESCRIPTION
**Pull request by srikanthv02**
…ocomplete single and multi

### CODE REVIEW

File: libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts [hunk 1]
```diff
@@ -88,6 +88,10 @@ export class AutocompleteMultiComponent<T> extends ControlComponent<T[]> impleme
   hint?: string;
   @Input()
   removeDisabledOptions = true;
+  @Input()
+  allowDropdownTrigger: boolean = false;
+  @Input()
+  optionType: "lookup" | 'raw' = "lookup";
```
Comment (srikanthv02) [inline libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts:94]: issue being, when using a autocomplete multi, it was adding "create" to the mat-chip

File: libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts [hunk 2]
```diff
@@ -88,6 +88,10 @@ export class AutocompleteMultiComponent<T> extends ControlComponent<T[]> impleme
   hint?: string;
   @Input()
   removeDisabledOptions = true;
+  @Input()
+  allowDropdownTrigger: boolean = false;
```
Comment (WilliamChelman) [inline libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts:92]:
```suggestion
  profile: 'search' | 'select' = 'search';
```

File: libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts (hunk 1 above)
Comment (WilliamChelman) [inline libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts:94]:
```suggestion
  cacheOptions: boolean = true;
```
and in that method
```ts
private async storeValueOption(value: T): Promise<SelectOption<T> | undefined> {
    if (!this._optionsProvider || !value) return undefined;
    const hasOption = await firstValueFrom(this.optionsProvider.hasOptionFor(value));
    if (!hasOption) return undefined;
    const option = await firstValueFrom(this._optionsProvider.getValueOption(value));
    if(this.cacheOptions) this.storedValueOptions.push(option);
    return option;
  }
```

File: libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts [hunk 3]
```diff
@@ -260,9 +264,11 @@ export class AutocompleteMultiComponent<T> extends ControlComponent<T[]> impleme
...
   private async getSelectOption(value: T): Promise<SelectOption<T> | undefined> {
     if (value == null) return undefined;
-    const allOptions = [...this.storedValueOptions, ...this.options];
-    const option = allOptions.find(o => o.value === value);
-    return option ?? this.storeValueOption(value);
+    if (this.optionType === 'lookup') {
+      const allOptions = [...this.storedValueOptions, ...this.options];
+      return allOptions.find(o => o.value === value);
+    }
+    return this.storeValueOption(value);
```
Comment (WilliamChelman) [inline libs/legi-shared/autocomplete/autocomplete-multi/autocomplete-multi.component.ts:271]: This change can be reverted

**Review by WilliamChelman — CHANGES_REQUESTED** [review]

**Review by WilliamChelman — APPROVED** [review]

**Pull request merged**